src/pages/SignUp/SignUp.js

```diff
@@ -1,10 +1,197 @@
 import React, { Component } from 'react';
+import './SignUp.scss';
 
 class SignUp extends Component {
+    constructor() {
+        super();
+        this.state = {
+            id: '',
+            pw: '',
+            name: '',
+            phone:'',
+            email: '',
+            passwordcheck: '',
+
+            validID: true,
+            validPW: true,
+            validDoubleCheck: true,
+            validName: true,
+            validPhone: true,
+            validEmail: true,
+        }
+    }
+    validationCheck = (type, value) => {
+        if (type==='id') {
+            let idReg = /^[a-z]+[a-z0-9]{5,19}$/g;
+            const validID = idReg.test(value);
+            this.setState({validID});
+        } else if (type==='pw') {
+            let passwordRules = /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,16}$/;
+            const validPW = passwordRules.test(value);
+            this.setState({validPW})
+        } else if (type === 'name') {
+            let reg = /^[가-힣]{2,4}$/;
+            const validName = reg.test(value);
+            this.setState({validName});
+        } else if (type === 'email') {
+            let emailRule = /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+            const validEmail = emailRule.test(value);
+            this.setState({validEmail})
+        }
+    }
+
+    handleInputValueChange = (e) => {
+        // const { id, pw, name, phone, email } = e.target;
+        this.validationCheck(e.target.name, e.target.value);
+        this.setState({[e.target.name]: e.target.value});
+    }
```

Comment:
```suggestion
    handleInputValueChange = (e) => {
        const { name, value } = e.target;
        this.validationCheck(name, value);
        this.setState({[name]: value});
    }
```
이렇게 비구조화 할당 가능하겠네요 :)

Reply:
네엡 수정했습니다 !